
This patch:

-reworks playback debug osd pop-up to more visualy nice looking onbe
-adds customized osdnavigation window with look&feel coresponding to Monochrome style


diff --speed-large-files --no-dereference --minimal -Naur Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/osd.xml Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/osd.xml
--- Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/osd.xml	2022-04-26 01:09:17.000000000 +0200
+++ Monochrome-8569391fd35dc01aaec19f8085b3002ede8148b2/osd.xml	2022-08-06 17:45:40.670904485 +0200
@@ -826,198 +784,939 @@
 
     <!--====================================================================-->
 
-
     <window name="osd_debug">
-        <area>0,100%-250,100%,250</area>
+        <fontdef name="normal" face="DejaVu LGC Sans">
+            <pixelsize>24</pixelsize>
+            <stretch>extracondensed</stretch>
+            <weight>normal</weight>
+            <color>#FFFFFF</color>
+            <shadowoffset>1,1</shadowoffset>
+            <shadowcolor>#000000</shadowcolor>
+        </fontdef>
+        <fontdef name="normal_tiny" from="normal">
+            <pixelsize>16</pixelsize>
+        </fontdef>
+        <fontdef name="title_small" face="Fluid">
+            <color>#FFFFFF</color>
+            <pixelsize>25</pixelsize>
+            <stretch>extracondensed</stretch>
+            <weight>normal</weight>
+            <shadowoffset>1,1</shadowoffset>
+            <shadowcolor>#000000</shadowcolor>
+        </fontdef>
+        <area>1620,200,300,780</area>
         <shape name="background">
             <area>0,0,100%,100%</area>
             <type>box</type>
             <fill color="#000000" alpha="255" />
             <line color="#000000" alpha="0" width="0" />
-            <alpha>200</alpha>
+            <alpha>210</alpha>
         </shape>
-        <imagetype name="separator_h">
-            <area>0,0,100%,1</area>
-            <filename>images/SeparatorH.png</filename>
+        <animation trigger="AboutToShow">
+            <section duration="500" centre="middle">
+                <position start="1280,0" end="968,0" easingcurve="OutBack" />
+            </section>
+        </animation>
+        <animation trigger="AboutToHide">
+            <section duration="500" centre="middle">
+                <position start="968,0" end="1280,0" easingcurve="InBack" />
+            </section>
+        </animation>
+        <imagetype name="popup_background">
+            <area>0,0,350,720</area>
+            <filename>images/shared/VisOsdPanel.png</filename>
         </imagetype>
 
-        <fontdef name="osdfont_s" from="basefont_s"/>
+        <imagetype name="storage-divider">
+            <area>0,25,316,35</area>
+            <filename>images/shared/MenuItemNF.png</filename>
+        </imagetype>
 
-        <textarea name="file">
-            <font>osdfont_s</font>
-            <area>50,25,175,40</area>
+        <imagetype name="storage_icon">
+            <area>32,70,45,45</area>
+            <filename>images/osd/disk_props.png</filename>
+        </imagetype>
+
+        <textarea name="storage">
+            <area>32,120,130,35</area>
+            <font>normal</font>
+            <multiline>no</multiline>
+            <cutdown>yes</cutdown>
+            <case>upper</case>
             <align>left,vcenter</align>
-            <value>Filename :</value>
+            <alpha>255</alpha>
+            <value>Storage to Buffer :</value>
         </textarea>
-        <textarea name="filename">
-            <font>osdfont_s</font>
-            <area>225,25,980,40</area>
+        <textarea name="storagerate">
+            <area>160,120,150,35</area>
+            <font>normal</font>
+            <multiline>no</multiline>
+            <cutdown>yes</cutdown>
+            <case>normal</case>
             <align>left,vcenter</align>
-            <alpha>200</alpha>
+            <alpha>128</alpha>
+        </textarea>
+        <textarea name="buffer" from="storage">
+            <position>32,155</position>
+            <value>Available Buffer :</value>
+        </textarea>
+        <textarea name="bufferavail" from="storagerate">
+            <position>160,155</position>
+            <template>%BUFFERAVAIL% of %BUFFERSIZE%Mb</template>
         </textarea>
 
-        <group name="audiodetails">
-            <area>50,75,350,150</area>
-            <textarea name="audio">
-                <font>osdfont_s</font>
-                <area>0,0,175,40</area>
-                <align>left,vcenter</align>
-                <value>Audio :</value>
-            </textarea>
-            <textarea name="audiocodec">
-                <font>osdfont_s</font>
-                <area>175,0,150,40</area>
-                <align>left,vcenter</align>
-                <alpha>200</alpha>
-            </textarea>
-            <textarea name="channels">
-                <font>osdfont_s</font>
-                <area>0,50,175,40</area>
-                <align>left,vcenter</align>
-                <value>Channels :</value>
-            </textarea>
-            <textarea name="audiochannels">
-                <font>osdfont_s</font>
-                <area>175,50,150,40</area>
-                <align>left,vcenter</align>
-                <alpha>200</alpha>
-            </textarea>
-            <textarea name="audiorate">
-                <font>osdfont_s</font>
-                <area>0,100,175,40</area>
-                <align>left,vcenter</align>
-                <value>Sample rate :</value>
-            </textarea>
-            <textarea name="samplerate">
-                <font>osdfont_s</font>
-                <area>175,100,150,40</area>
-                <align>left,vcenter</align>
-                <alpha>200</alpha>
-            </textarea>
-        </group>
+        <imagetype name="video-divider">
+            <area>0,162,316,35</area>
+            <filename>images/shared/MenuItemNF.png</filename>
+        </imagetype>
 
-        <group name="videodetails">
-            <area>450,75,450,150</area>
-            <textarea name="video">
-                <font>osdfont_s</font>
-                <area>0,0,175,40</area>
-                <align>left,vcenter</align>
-                <value>Video :</value>
-            </textarea>
-            <textarea name="videowidth">
-                <font>osdfont_s</font>
-                <area>175,0,250,40</area>
-                <align>left,vcenter</align>
-                <template>%VIDEOWIDTH%x%VIDEOHEIGHT%@%VIDEOFRAMERATE%fps</template>
-                <alpha>200</alpha>
-            </textarea>
-            <textarea name="vidcodec">
-                <font>osdfont_s</font>
-                <area>0,50,175,40</area>
-                <align>left,vcenter</align>
-                <value>Codec/Dec :</value>
-            </textarea>
-            <textarea name="videocodec">
-                <font>osdfont_s</font>
-                <area>175,50,250,40</area>
-                <align>left,vcenter</align>
-                <template>%VIDEOCODEC% %VIDEODECODER%</template>
-                <alpha>200</alpha>
-            </textarea>
-            <textarea name="cpuload">
-                <font>osdfont_s</font>
-                <area>0,100,175,40</area>
-                <align>left,vcenter</align>
-                <value>CPUs :</value>
-            </textarea>
-            <textarea name="load">
-                <font>osdfont_s</font>
-                <area>175,100,250,40</area>
-                <align>left,vcenter</align>
-                <alpha>200</alpha>
-            </textarea>
+        <imagetype name="video_icon">
+            <area>32,210,45,45</area>
+            <filename>images/osd/video_props.png</filename>
+        </imagetype>
+
+        <textarea name="decoder" from="storage">
+            <position>32,260</position>
+            <value>Buffer to Decoder :</value>
+        </textarea>
+        <textarea name="decoderrate" from="storagerate">
+            <position>160,260</position>
+        </textarea>
+
+        <textarea name="video" from="storage">
+            <position>32,295</position>
+            <value>Video :</value>
+        </textarea>
+        <textarea name="videowidth" from="storagerate">
+            <position>160,295</position>
+            <template>%VIDEOWIDTH%x%VIDEOHEIGHT%@%VIDEOFRAMERATE%</template>
+        </textarea>
+        <textarea name="vidcodec" from="storage">
+            <position>32,330</position>
+            <value>Codec/Dec :</value>
+        </textarea>
+        <textarea name="videocodec" from="storagerate">
+            <position>160,330</position>
+            <template>%VIDEOCODEC%/%VIDEODECODER%</template>
+        </textarea>
+        <textarea name="deint" from="storage">
+            <position>32,365</position>
+            <value>Deint :</value>
+        </textarea>
+        <textarea name="deinterlacer" from="storagerate">
+            <position>160,365</position>
+        </textarea>
+        <textarea name="cpuload" from="storage">
+            <position>32,400</position>
+            <value>CPUs :</value>
+        </textarea>
+        <textarea name="load" from="storagerate">
+            <position>160,400</position>
+        </textarea>
+
+        <textarea name="fps" from="storage">
+            <position>32,435</position>
+            <value>FPS :</value>
+        </textarea>
+        <textarea name="framerate" from="storagerate">
+            <position>160,435</position>
+        </textarea>
+
+        <textarea name="sync" from="storage">
+            <position>32,470</position>
+            <value>A/V Sync :</value>
+        </textarea>
+        <textarea name="avsync" from="storagerate">
+            <position>160,470</position>
+        </textarea>
+        <textarea name="frames" from="storage">
+            <position>32,505</position>
+            <value>Frames decoded/free :</value>
+        </textarea>
+        <textarea name="videoframes" from="storagerate">
+            <position>160,505</position>
+            <template>%VIDEOFRAMES% frames</template>
+        </textarea>
+
+        <imagetype name="audio-divider">
+            <area>0,507,316,35</area>
+            <filename>images/shared/MenuItemNF.png</filename>
+        </imagetype>
+
+        <imagetype name="audio_icon">
+            <area>32,550,45,45</area>
+            <filename>images/osd/audio_props.png</filename>
+        </imagetype>
+
+        <textarea name="audio" from="storage">
+            <position>32,600</position>
+            <value>Audio :</value>
+        </textarea>
+        <textarea name="audiocodec" from="storagerate">
+            <position>160,600</position>
+        </textarea>
+        <textarea name="channels" from="storage">
+            <position>32,635</position>
+            <value>Channels :</value>
+        </textarea>
+        <textarea name="audiochannels" from="storagerate">
+            <position>160,635</position>
+        </textarea>
+        <textarea name="audiorate" from="storage">
+            <position>32,670</position>
+            <value>Sample rate :</value>
+        </textarea>
+        <textarea name="samplerate" from="storagerate">
+            <position>160,670</position>
+        </textarea>
+
+        <imagetype name="end-divider">
+            <area>0,677,316,35</area>
+            <filename>images/shared/MenuItemNF.png</filename>
+        </imagetype>
+
+    </window>
+
+    <window name="osd_navigation">
+        <font>basefont_l</font>
+        <area>62,260,675,80,800x600</area>
+        <shape name="background">
+            <area>0,-30,100%,85%</area>
+            <type>roundbox</type>
+            <fill color="#000000" alpha="200" />
+            <line color="#222222" alpha="255" width="2" />
+            <cornerradius>12</cornerradius>
+        </shape>
+
+        <button name="more">
+            <focusorder>2</focusorder>
+            <area>6,0,112,75,800x600</area>
+            <statetype name="buttonstate">
+                <state name="active">
+                    <imagetype name="icon">
+                         <!-- <area>46,5,20,20,800x600 </area> -->
+                         <area>74,6,32,32,1280x720</area>
+                        <filename>images/icons/Playlist.png</filename>
+                    </imagetype>
+                </state>
+                <state name="selected">
+                    <shape name="background">
+                        <!-- <area>44,0,24,30,800x600 </area> -->
+                        <area>68,0,45,45,1280x720</area>
+                        <type>box</type>
+                        <fill color="#999999" alpha="200" />
+                    </shape>
+                    <imagetype name="icon">
+                         <!-- <area>46,5,20,20,800x600 </area> -->
+                         <area>74,6,32,32,1280x720</area>
+                        <filename>images/icons/Playlist.png</filename>
+                    </imagetype>
+                    <textarea name="text">
+                        <!-- <area>0,22,112,50,800x600 </area> -->
+                        <area>0,23,179,78,1280x720</area>
+                        <align>allcenter</align>
+                        <font>basefont_l</font>
+                        <cutdown>none</cutdown>
+                        <value>More Actions</value>
+                    </textarea>
+                </state>
+                <state name="disabled" from="active" />
+                <state name="pushed" from="selected" />
+            </statetype>
+        </button>
+
+        <group name="grp0">
+            <area>0,0,675,80,800x600</area>
+
+            <button name="MENU">
+                <focusorder>2</focusorder>
+                <area>75,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Profile.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Profile.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Menu</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="JUMPRWND">
+                <focusorder>2</focusorder>
+                <area>144,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Rewind.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Rewind.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Back more</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="SEEKRWND">
+                <focusorder>2</focusorder>
+                <area>213,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Prev.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Prev.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Jump back</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="PLAY">
+                <area>282,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Play.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Play.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Play</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="PAUSE">
+                <area>282,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Pause.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Pause.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Pause</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="SEEKFFWD">
+                <area>351,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Next.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Next.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Jump fwd</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="JUMPFFWD">
+                <area>420,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Forward.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Forward.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Fwd more</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="INFO">
+                <area>489,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Override.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Override.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Information</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="MUTE">
+                <area>558,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="mutebn">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Mute.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Mute.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Mute</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="unmute">
+                <area>558,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Mute.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Mute.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Unmute</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
         </group>
 
-        <group name="processingdetails">
-            <area>950,75,500,150</area>
-            <textarea name="fps">
-                <font>osdfont_s</font>
-                <area>0,0,275,40</area>
-                <align>left,vcenter</align>
-                <value>FPS :</value>
-            </textarea>
-            <textarea name="framerate">
-                <font>osdfont_s</font>
-                <area>275,0,200,40</area>
-                <align>left,vcenter</align>
-                <alpha>200</alpha>
-            </textarea>
-            <textarea name="sync">
-                <font>osdfont_s</font>
-                <area>0,50,275,40</area>
-                <align>left,vcenter</align>
-                <value>A/V Sync :</value>
-            </textarea>
-            <textarea name="avsync">
-                <font>osdfont_s</font>
-                <area>275,50,200,40</area>
-                <align>left,vcenter</align>
-                <alpha>200</alpha>
-            </textarea>
-            <textarea name="frames">
-                <font>osdfont_s</font>
-                <area>0,100,275,40</area>
-                <align>left,vcenter</align>
-                <value>Frames decoded/free :</value>
-            </textarea>
-            <textarea name="videoframes">
-                <font>osdfont_s</font>
-                <area>275,100,200,40</area>
-                <align>left,vcenter</align>
-                <alpha>200</alpha>
-            </textarea>
+        <group name="grp1">
+            <area>0,0,675,80,800x600</area>
+
+            <button name="TOGGLEFILL">
+                <area>75,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Corner.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Corner.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Fill</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />200
+                </statetype>
+            </button>
+
+            <button name="TOGGLEASPECT">
+                <area>144,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>70,12,58,24,1280x720</area>
+                            <filename>images/icons/VideoWidescreen.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>70,12,58,24,1280x720</area>
+                            <filename>images/icons/VideoWidescreen.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Aspect Ratio</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="NEXTCC">
+                <area>213,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>70,12,56,24,1280x720</area>
+                            <filename>images/icons/SubCC.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>70,12,56,24,1280x720</area>
+                            <filename>images/icons/SubCC.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Captions</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="JUMPSTART">
+                <area>282,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Home.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Home.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Jump to start</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="SETBOOKMARK">
+                <area>351,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Cutlist.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Cutlist.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Bookmark</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="JUMPBKMRK">
+                <area>420,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Bookmark.png</filename>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>images/icons/Bookmark.png</filename>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Jump bkmark</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
         </group>
-        <group name="storagedetails">
-            <area>1450,75,500,150</area>
-            <textarea name="storage">
-                <font>osdfont_s</font>
-                <area>0,0,250,40</area>
-                <align>left,vcenter</align>
-                <value>Storage to Buffer :</value>
-            </textarea>
-            <textarea name="storagerate">
-                <font>osdfont_s</font>
-                <area>250,0,250,40</area>
-                <align>left,vcenter</align>
-                <alpha>200</alpha>
-            </textarea>
-            <textarea name="decoder">
-                <font>osdfont_s</font>
-                <area>0,50,250,40</area>
-                <align>left,vcenter</align>
-                <value>Buffer to Decoder :</value>
-            </textarea>
-            <textarea name="decoderrate">
-                <font>osdfont_s</font>
-                <area>250,50,250,40</area>
-                <align>left,vcenter</align>
-                <alpha>200</alpha>
-            </textarea>
-            <textarea name="buffer">
-                <font>osdfont_s</font>
-                <area>0,100,250,40</area>
-                <align>left,vcenter</align>
-                <value>Available Buffer :</value>
-            </textarea>
-            <textarea name="bufferavail">
-                <font>osdfont_s</font>
-                <area>250,100,250,40</area>
-                <align>left,vcenter</align>
-                <template>%BUFFERAVAIL% of %BUFFERSIZE%Mb</template>
-                <alpha>200</alpha>
-            </textarea>
+
+        <group name="grp2">
+            <area>0,0,675,80,800x600</area>
+
+            <button name="RWNDSTICKY">
+                <focusorder>2</focusorder>
+                <area>213,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>rew_button.png</filename>
+                            <grayscale>true</grayscale>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>rew_button.png</filename>
+                            <grayscale>true</grayscale>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Rewind</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="PLAY#">
+                <area>282,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>play_button.png</filename>
+                            <grayscale>true</grayscale>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>68,0,45,45,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>play_button.png</filename>
+                            <grayscale>true</grayscale>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Play</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
+
+            <button name="FFWDSTICKY">
+                <area>351,0,112,75,800x600</area>
+                <statetype name="buttonstate">
+                    <state name="active">
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>ff_button.png</filename>
+                            <grayscale>true</grayscale>
+                        </imagetype>
+                    </state>
+                    <state name="selected">
+                        <shape name="background">
+                            <area>70,0,38,46,1280x720</area>
+                            <type>box</type>
+                            <fill color="#999999" alpha="200" />
+                        </shape>
+                        <imagetype name="icon">
+                            <area>74,6,32,32,1280x720</area>
+                            <filename>ff_button.png</filename>
+                            <grayscale>true</grayscale>
+                        </imagetype>
+                        <textarea name="text">
+                            <area>0,23,179,78,1280x720</area>
+                            <align>allcenter</align>
+                            <font>basefont_l</font>
+                            <cutdown>none</cutdown>
+                            <value>Fast fwd</value>
+                        </textarea>
+                    </state>
+                    <state name="disabled" from="active" />
+                    <state name="pushed" from="selected" />
+                </statetype>
+            </button>
         </group>
+    </window>
 
+    <window name="program_info">
+        <fontdef name="small" face="DejaVu Sans">
+            <pixelsize>22</pixelsize>
+            <color>#FFFFFF</color>
+            <shadowoffset>1,1</shadowoffset>
+            <shadowcolor>#000000</shadowcolor>
+        </fontdef>
+        <fontdef name="smaller" from="small">
+            <pixelsize>18</pixelsize>
+        </fontdef>
+        <fontdef name="medium" from="small">
+            <pixelsize>26</pixelsize>
+        </fontdef>
+        <area>100,370,1080,300</area>
+        <shape name="background">
+            <area>0,0,100%,100%</area>
+            <type>roundbox</type>
+            <fill color="#000000" alpha="200" />
+            <line color="#222222" alpha="255" width="2" />
+            <cornerradius>12</cornerradius>
+        </shape>
+        <textarea name="title">
+            <font>medium</font>
+            <area>10,10,600,30</area>
+            <align>left,top</align>
+        </textarea>
+        <textarea name="starttime">
+            <font>smaller</font>
+            <area>10,50,700,30</area>
+            <align>left,top</align>
+            <template>%STARTTIME%-%ENDTIME% : %LENMINS%</template>
+        </textarea>
+        <clock name="clock">
+            <area>770,10,300,30</area>
+            <font>small</font>
+            <template>%TIME%</template>
+            <align>right,top</align>
+        </clock>
+        <textarea name="channum">
+            <font>small</font>
+            <area>20,100,150,30</area>
+            <align>hcenter,vcenter</align>
+        </textarea>
+        <imagetype name="iconpath" depends="channum">
+            <area>30,140,130,100</area>
+        </imagetype>
+        <imagetype name="coverart" depends="!channum">
+            <area>45,100,100,150</area>
+        </imagetype>
+        <textarea name="callsign" depends="channum">
+            <font>smaller</font>
+            <area>20,250,150,30</area>
+            <align>hcenter,vcenter</align>
+        </textarea>
+        <textarea name="description">
+            <font>small</font>
+            <area>200,100,860,180</area>
+            <align>left,top</align>
+            <multiline>yes</multiline>
+            <template>%"|SUBTITLE|" %%LONGREPEAT%%(|STARS|) %%DESCRIPTION%</template>
+        </textarea>
     </window>
 
 </mythuitheme>
